docs/ImpactAnalysis.md

```diff
@@ -28,12 +28,130 @@ This document is the first iteration of a high-level design document for the Lei
 
 # Ecosystem
 
+Most users interact with the Cardano system through a wallet or a dApp. Both of these systems provide end-user interfaces and are directly or indirectly connected to the Cardano network. A varying number of services in between the Cardano network and those user-facing applications determines the impact of the Leios upgrade onto the Cardano ecosystem. The Cardano network itself is operated by so-called stake pool operators (SPOs). The following context diagram illustrates notable personas, a few example systems of the Cardano ecosystem and their interactions:
+
+![](./context-diagram.svg)
+
+## Impact on operations
+
+The Leios upgrade does not change the roles and responsibilities for SPOs and only small changes to their operations are expected. Besides a modest increase in compute and network requirements (see [resource requirements summary in the CIP](https://github.com/cardano-scaling/CIPs/blob/leios/CIP-0164/README.md#resource-requirements)), one additional step in the block producing node setup is required: One additional key pair is needed to sign votes for EBs.
+
+- **REQ-GenerateBLSKeys** SPOs must be able to generate BLS key pairs.
+- **REQ-RegisterBLSKeys** SPOs must be able to register their BLS public key to become part of the voting committee.
+- **REQ-RotateBLSKeys** SPOs must be able to rotate their BLS key and limit usage of compromised keys.
+
+Concretely, these BLS keys could be generated and managed alongside the existing VRF keys. The node configuration needs to be extended to include the BLS signing key and the public key will need to become part of the operational certificate. With the signing key available, the node will be able to automatically issue votes for EBs.
```

Comment:
> public key will need to become part of the operational certificate.

The opcert is part of the Praos header. So does this line imply the Praos header needs to include an entire public key? That'd be a non-trivial increase. And it's not listed here https://github.com/cardano-scaling/CIPs/blob/leios/CIP-0164/README.md#ranking-blocks-rbs

So maybe this is miswording?

Reply:
Good catch. As you can tell, I'm not (yet) very familiar with the opcert situation. Having the BLS public key in each block header is probably not needed.. so where should it go, how should registration be achieved? Maybe we don't know and I'll just drop the sentence?